qcelemental/models/procedures.py

```diff
@@ -60,3 +82,27 @@ class Optimization(OptimizationInput):
         ..., description="The success of a given programs execution. If False, other fields may be blank.")
     error: Optional[ComputeError] = Schema(None, description=str(ComputeError.__doc__))
     provenance: Provenance = Schema(..., description=str(Provenance.__doc__))
+
+    @validator('trajectory', whole=True)
+    def _trajectory_protocol(cls, v, values):
+
+        # Do not propogate validation errors
+        if 'protocols' not in values:
+            raise ValueError("Protocols was not properly formed.")
+
+        keep_enum = values['protocols'].trajectory
+        if keep_enum == 'all':
+            pass
+        elif keep_enum == "initial_and_final":
+            if len(v) != 2:
+                v = [v[0], v[-1]]
+        elif keep_enum == "final":
+            if len(v) != 1:
+                v = [v[-1]]
+        elif keep_enum == "none":
```

Comment:
should this and above also set "initial_molecule" and "final_molecule" to None as appropriate? Or does that happen later?

Reply:
Modifying these would be different fields. `protocols: initial_molecule`.